### DISCUSSION TITLE
How to use multiprocessing at all?

### DISCUSSION THREAD

**Discussion by rubslopes** [kernc/backtesting.py#464] [Q&A] [answered]
Is there a tutorial? I searched a lot but I couldn't find one. But running the optimizer gives the message `For multiprocessing support in Backtest.optimize() set multiprocessing start method to 'fork'.`

I am a Mac user.

Thank you!

**Comment by kernc** [comment, marked as answer]
See [Contexts and Start Methods](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods) in Python `multiprocessing` docs.

**Reply by rubslopes** [reply]
Thanks!

For anyone wondering, simply include this in your code:

```python
import multiprocessing as mp
mp.set_start_method('fork')
```

This only works on Unix systems.

**Comment by stnatter** [comment]
tried this. getting: `RuntimeError: context has already been set`

**Reply by rubslopes** [reply]
You should run `mp.set_start_method('fork')` only once when importing. Don't run it again.

**Comment by curiouscod3** [comment]
```python
import multiprocessing as mp

try:
    # mp.set_start_method("spawn", force=True)
    # print("spawned")
    mp.set_start_method("fork", force=True)
    print("forked")
except RuntimeError:
    pass
```

**Reply by lILogit** [reply]
It works also on iOS (MacBook Pro)
